tests/drivers/i2c/i2c_target_api/testcase.yaml

```diff
@@ -32,3 +32,10 @@ tests:
       - nucleo_f091rc
     extra_configs:
       - CONFIG_APP_DUAL_ROLE_I2C=y
+  drivers.i2c.target_api.single_role:
```

Comment:
besides, look at the i2c_mcux_lpi2c.c if the CONFIG_I2C_TARGET is defined, it can support dual mode. why only enable single mode. @danieldegrasse can you confirm?